detector.py

```diff
@@ -375,29 +383,40 @@ def saveDetectionsToFile(self, path):
         Writes current detections to a file at path. Values are separated by ';'.
         """
 
-        # Default formatting
-        f1 = "{:.5f}"
-        lineBase1 = "{};" + f"{f1};{f1};{f1};{f1};"
-
         try:
-            with open(path, "w") as file:
-                file.write(
-                    "frame;length;distance;angle;aspect;"
-                    "corner1 x;corner1 y;corner2 x;corner2 y;"
-                    "corner3 x;corner3 y;corner4 x;corner4 y\n"
-                )
-                for frame, dets in enumerate(self.detections):
-                    if dets is not None:
-                        for d in dets:
-                            if d.corners is not None:
-                                file.write(
-                                    lineBase1.format(
-                                        frame, d.length, d.distance, d.angle, d.aspect
-                                    )
+            rows = []
+            for frame, dets in enumerate(self.detections):
+                if dets is not None:
```

Comment:
Why dets and fps and frame_pc_time should be None?

To avoid extra indentation within for loops, you can write `if not dets: continue`